Bottom: 82f974ac7d1ab440fac536597e73a6cea75af80d
Top:    5ed7e9bb453abfd639b3b196f4417d8e72668861
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-09-20 09:53:42 -0500

Refresh of add-haproxy-role

---
diff --git a/ansible-playbook/group_vars/docker-hosts/networking.yml b/ansible-playbook/group_vars/docker-hosts/networking.yml
index 32d99802b..977d118e1 100644
--- a/ansible-playbook/group_vars/docker-hosts/networking.yml
+++ b/ansible-playbook/group_vars/docker-hosts/networking.yml
@@ -22,14 +22,14 @@ testssh_port_args:
 
 exim4_hostname: mail.novatech-llc.com
 exim4_port_args:
-- "25:25"
-- "465:465"
-- "587:587"
+- "{{ ansible_facts.default_ipv4.address }}:25:25"
+- "{{ ansible_facts.default_ipv4.address }}:465:465"
+- "{{ ansible_facts.default_ipv4.address }}:587:587"
 
 openldap_hostname: ldap.novatech-llc.com
 openldap_port_args:
-- "389:389"
-- "636:636"
+- "{{ ansible_facts.default_ipv4.address }}:389:389"
+- "{{ ansible_facts.default_ipv4.address }}:636:636"
 
 phpldapadmin_hostname: ldap.novatech-llc.com
 phpldapadmin_port_args: []
@@ -38,12 +38,12 @@ phpmyadmin_hostname: phpmyadmin.novatech-llc.com
 
 buildbot_hostname: buildbot.novatech-llc.com
 buildbot_port_args:
-- "9989:9989"
+- "{{ ansible_facts.default_ipv4.address }}:9989:9989"
 
 nexus_hostname: nexus.novatech-llc.com
 nexus_port_args: []
 
 haproxy_port_args:
-- "80:80"
-- "443:443"
-- "9000:9000"
+- "{{ ansible_facts.default_ipv4.address }}:80:80"
+- "{{ ansible_facts.default_ipv4.address }}:443:443"
+- "{{ ansible_facts.default_ipv4.address }}:9000:9000"
diff --git a/roles/haproxy/tasks/main.yml b/roles/haproxy/tasks/main.yml
index 63a42aff4..533ab9c3d 100644
--- a/roles/haproxy/tasks/main.yml
+++ b/roles/haproxy/tasks/main.yml
@@ -4,6 +4,17 @@
     that:
     - haproxy_stats_password is defined
 
+# Setup the configuration directory
+- name: docker_container.conf dir
+  file:
+    state: directory
+    path: '{{ config_hostdir }}'
+    owner: root
+    group: root
+    mode: 'u=rwx,g=rx,o=rx'
+    recurse: no
+
+
 - name: Populate config files
   template:
     src: "{{ item }}.j2"
@@ -19,5 +30,6 @@
     image: '{{ image_name }}'
     networks: '{{ container_networks }}'
     ports: '{{ container_port_args }}'
+    purge_networks: true
     state: started
     volumes: '{{ container_volumes }}'
diff --git a/roles/haproxy/templates/haproxy.cfg.j2 b/roles/haproxy/templates/haproxy.cfg.j2
index aebe31d01..228ef73a0 100644
--- a/roles/haproxy/templates/haproxy.cfg.j2
+++ b/roles/haproxy/templates/haproxy.cfg.j2
@@ -87,7 +87,7 @@ backend backend-testssh
     server testssh1 testssh.frontend:80
 
 backend backend-buildbot
-    server buildbot1 buildbot.frontend:80
+    server buildbot1 buildbot.frontend:8080
 
 backend backend-nexus
     server nexus1 nexus3.frontend:8081
